lora-phy/src/lib.rs

```diff
@@ -275,9 +275,32 @@ where
         }
     }
 
+    /// Prepare radio to listen to a given frequency
+    pub async fn prepare_for_listen(&mut self, frequency_in_hz: u32) -> Result<(), RadioError> {
+        self.prepare_modem(frequency_in_hz).await?;
+        self.radio_kind.set_channel(frequency_in_hz).await?;
+        self.radio_mode = RadioMode::Listen;
+        Ok(())
+    }
+
+    /// Start listening
+    pub async fn listen(&mut self) -> Result<(), RadioError> {
+        if self.radio_mode == RadioMode::Listen {
+            self.radio_kind.do_rx(RxMode::Continuous).await?;
+            Ok(())
+        } else {
+            Err(RadioError::InvalidRadioMode)
+        }
+    }
+
+    /// Get the current rssi
+    pub async fn get_rssi(&mut self) -> Result<i16, RadioError> {
+        self.radio_kind.get_rssi().await
```

Comment:
Shouldn't we check whether we are in correct mode?

Reply:
I skipped mode checking because I though someone might want to use the function regardless of the current mode, such as checking the last RSSI (not the packet one) after performing rx() is done or timed out. It would be unable to do such things If we restrict it to listen mode only and remove `prepare_listen`, but I personally don't mind either way.

Reply:
Ok, makes sense!